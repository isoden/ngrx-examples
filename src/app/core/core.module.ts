import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';

import { AppComponent } from './containers/app/app.component';
import { TopPageComponent } from './containers/top-page/top-page.component';

@NgModule({
  imports: [
    CommonModule,
    RouterModule.forChild([
      {
        path     : '',
        component: TopPageComponent,
      },
    ]),
  ],
  declarations: [
    AppComponent,
    TopPageComponent,
  ],
})
export class CoreModule { }
