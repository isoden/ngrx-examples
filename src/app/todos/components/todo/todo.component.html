<p>
  <label>
    <input type="checkbox" [checked]="todo.completed" (change)="onToggle()">
    <span class="sr-only">Toggle complete</span>
  </label>
  <span [style.text-decoration]="todo.completed ? 'line-through' : ''">{{ todo.title }}</span>
</p>
