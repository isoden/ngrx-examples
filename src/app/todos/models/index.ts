export { Todo } from './todo';
export { VisibilityFilter } from './visibility-filter';
