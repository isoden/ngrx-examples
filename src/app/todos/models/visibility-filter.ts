export type VisibilityFilter = 'SHOW_ALL' | 'SHOW_ACTIVE' | 'SHOW_COMPLETED';
