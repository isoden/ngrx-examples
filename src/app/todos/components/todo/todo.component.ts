import { Component, OnInit, ChangeDetectionStrategy, Input, Output, EventEmitter, HostBinding } from '@angular/core';
import { Todo } from '../../models';

@Component({
  selector: 'app-todo',
  templateUrl: './todo.component.html',
  styleUrls: ['./todo.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class TodoComponent implements OnInit {
  @Input()
  todo: Todo;

  @Output()
  toggle = new EventEmitter<number>();

  constructor() { }

  ngOnInit() {
  }

  onToggle() {
    this.toggle.emit(this.todo.id);
  }
}
