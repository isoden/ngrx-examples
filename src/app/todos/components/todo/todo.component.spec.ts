import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { TodoComponent } from './todo.component';
import { ChangeDetectionStrategy } from '@angular/core';
import { Todo } from '../../models/index';

describe('TodoComponent', () => {
  let component: TodoComponent;
  let fixture: ComponentFixture<TodoComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [
        TodoComponent,
      ],
    })
    .overrideComponent(TodoComponent, {
      set: { changeDetection: ChangeDetectionStrategy.Default },
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(TodoComponent);
    component = fixture.componentInstance;
    component.todo = new Todo(0, 'Learn Angular');
    fixture.detectChanges();
  });

  it('should compile', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should display todo title', () => {
    component.todo = new Todo(0, 'Learn Zone.js');
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should render title with strikethrough if todo is completed', () => {
    component.todo = new Todo(0, 'Learn RxJS', true);
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should emit toggle on checked', () => {
    jest.spyOn(component.toggle, 'emit');
    component.onToggle();
    expect(component.toggle.emit).toHaveBeenCalledWith(component.todo.id);
  });
});
