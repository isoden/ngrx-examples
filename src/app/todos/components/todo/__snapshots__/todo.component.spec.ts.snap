// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoComponent should compile 1`] = `
<app-todo
  todo={[Function Todo]}
  toggle={[Function EventEmitter]}
>
  <p>
    
    
    <label>
      
      
      <input
        type="checkbox"
      />
      
      
      <span
        class="sr-only"
      >
        Toggle complete
      </span>
      
    
    </label>
    
    
    <span>
      Learn Angular
    </span>
    
  
  </p>
  
</app-todo>
`;

exports[`TodoComponent should display todo title 1`] = `
<app-todo
  todo={[Function Todo]}
  toggle={[Function EventEmitter]}
>
  <p>
    
    
    <label>
      
      
      <input
        type="checkbox"
      />
      
      
      <span
        class="sr-only"
      >
        Toggle complete
      </span>
      
    
    </label>
    
    
    <span>
      Learn Zone.js
    </span>
    
  
  </p>
  
</app-todo>
`;

exports[`TodoComponent should render title with strikethrough if todo is completed 1`] = `
<app-todo
  todo={[Function Todo]}
  toggle={[Function EventEmitter]}
>
  <p>
    
    
    <label>
      
      
      <input
        type="checkbox"
      />
      
      
      <span
        class="sr-only"
      >
        Toggle complete
      </span>
      
    
    </label>
    
    
    <span
      style="text-decoration: line-through;"
    >
      Learn RxJS
    </span>
    
  
  </p>
  
</app-todo>
`;
